### DISCUSSION TITLE
Errors when using "mix ef-sqlite" with "web" or "single" Templates

### DISCUSSION THREAD

**Discussion by Ryan-Britton** [ServiceStack/Discuss#122] [Q&A]
Hi there,

I am trying to get a simple multi-project microservice prototype up and running before committing to an indie license, but struggled to get up and running with a "single" or "web" template. I thought maybe I was missing something, and wanted to share my experience (or at the very least provide something to help other folks who run into the same problems):

if I go to https://www.servicestack.net/start and select a "web" or "single" template (enabling only SqlLite RDBMS support), like so: 

![image](https://github.com/ServiceStack/Discuss/assets/124330161/46e1784e-4338-4183-ab06-7b74f217cf21)

 the project which I download doesn't build, and has the following errors: 
 
 
![image](https://github.com/ServiceStack/Discuss/assets/124330161/16f2fcf0-0ac5-4120-9d5c-3a2a59980c41)

If I copy the "Data" folder and "Roles.cs" file which are generated when generating the "blazor-wasm" template:

![image](https://github.com/ServiceStack/Discuss/assets/124330161/d5ec2d4f-1e2e-44ad-ab81-70859d8e8cf7)

and add the ASP.NET Core Identity nuget packages (which are missing):

![image](https://github.com/ServiceStack/Discuss/assets/124330161/254376df-1d9b-467f-b2da-032c9d4747a7)

and add the ServiceStack.Server nuget (required for the AdminDatabase Feature)

```
 // Enable built-in Database Admin UI at /admin-ui/database
 services.AddPlugin(new AdminDatabaseFeature());
```

then I can get past the build errors - but I get the following error when trying to run "dotnet run --AppTasks=migrate"

![image](https://github.com/ServiceStack/Discuss/assets/124330161/5d4c248d-e50c-4fea-a0f3-2c8f02b19f00)

adding the following to Program.cs resolves this : 

```
services.AddAuthentication(options =>
    {
        options.DefaultScheme = IdentityConstants.ApplicationScheme;
        options.DefaultSignInScheme = IdentityConstants.ExternalScheme;
    })
    .AddIdentityCookies();
services.AddDataProtection()
    .PersistKeysToFileSystem(new DirectoryInfo("App_Data"));

services.AddIdentityCore<ApplicationUser>(options => options.SignIn.RequireConfirmedAccount = true)
    .AddRoles<IdentityRole>()
    .AddEntityFrameworkStores<ApplicationDbContext>()
    .AddSignInManager()
    .AddDefaultTokenProviders();
```

I realize that these steps might be considered "ASP.NET Core Identity Setup", and may differ per project...but I felt that it might be a better developer experience to have a project that builds and is ready to run (with some "simple defaults"), rather than something which appears to be "broken" out of the box....and so I wanted to make that suggestion here.

I also wanted to leave the step-by-step resolution here in case anyone else struggles with the broken templates....

**Comment by Ryan-Britton** [comment]
ADDITIONAL ISSUE : 

if I run `x mix -f ef-postgres` on the working project above (to switch from sqlite to posgres) - then I get the following error when running 'dotnet run --AppTasks=migrate' : 

![image](https://github.com/ServiceStack/Discuss/assets/124330161/c6eebde7-ed11-428f-b5b9-57cf9b59f543)

this happens when running against a NEW database (i.e.: there were no existing tables in the database before running "migrate" for the first time). Any help that can be offered in resolving this would be appreciated: I haven't managed to resolve this yet :-(

**Comment by Ryan-Britton** [comment]
I created a Git repository with the failing project source, for reference : 

[https://github.com/Ryan-Britton/ServiceStack-Issue-Example](https://github.com/Ryan-Britton/ServiceStack-Issue-Example)

**Comment by mythz** [comment]
I don't want to spend too much of my time on this, but I'll leave a couple of points:

 - The DB Mix templates are designed for adding DB to Identity Auth templates configured, as it needs to reference existing classes and replace EF Migration classes
 - If you want OrmLite with IdentityAuth start from one of the Identity Auth templates, if you want to start from scratch you're going to have to configure everything yourself, using the existing templates as a guide should help get to a working configuration.

If you just OrmLite on an empty template, use `sqlite` or `postgres` mix instead:

```sh
$ x mix sqlite
```
